src/MSBuildTaskHost/ChangeWavesStub.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// Stub implementation of ChangeWaves, that always returns true for AreFeaturesEnabled.
+    /// It is used to stub out the real ChangeWaves class, which is not available in the TaskHost.
+    /// </summary>
+    internal static class ChangeWaves
```

Comment:
This feels wrong...we can't disable versions on task hosts? Perhaps we haven't needed to before, but if we translate exceptions (or expect different translations), wouldn't that lead to bugs here whenever someone uses a task host? I looked at the ChangeWaves class, and it should be able to be compiled into MSBuildTaskHost with just one little helper method for Version.TryParse, something like:

```C#
        private static bool TryParseVersion(string stringVersion, out Version version)
        {
#if FEATURE_NET35_TASKHOST
            try
            {
                version = new Version(stringVersion);
                return true;
            }
            catch (Exception)
            {
                version = null;
                return false;
            }
#else
            return Version.TryParse(stringVersion, out version);
#endif
        }
```

That's about the same amount of code as this but cleaner and more correct as far as I can tell. Can we do that instead?

Reply:
Great idea, ready-made code to apply ... well - I love this suggestion! :-) Thanks!